### PR TITLE
[TCL30-44] Get rid of the subsequent reloading of the map

### PR DESCRIPTION
**Pull request by alxmcr**
## Description

When you drag our map, we reload our map, call to API with another map center location, and get new pins around the new center map.

## Related Issue

Closes [TCL30-44](https://the-collab-lab.atlassian.net/browse/TCL30-44)

## Acceptance Criteria

- If you drag our map, the map shouldn't be re-loaded.
- If you drag our map, our app should call API, and display new pins around to the new map center position.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|    | :sparkles: New feature     |
| ✓    | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

-  If you drag our map, the map should be re-loaded, and display the pins around the new map center position.

### After

-  If you drag our map, the map shouldn't be re-loaded, and display the pins around the new map center position.

## Testing Steps / QA Criteria

1. Clone repository: `git clone https://github.com/the-collab-lab/tcl-30-whats-near-me`.
2. Change to PR branch: `git checkout jw-ac-mm-share-location-button`
3. Install NPM dependencies: `npm install`.
4. Run dev server: `npm run start`
5. Open your browser: `http://localhost:3000/`
6. Drag the map, and you should see new pins around the new map center position, but our map should be reloaded.

### CODE REVIEW

File: src/hooks/useLocations.js [hunk 1]
```diff
@@ -26,8 +26,10 @@ export const useLocations = (lat, lng) => {
         setStatus('success');
       })
       .catch((err) => {
-        setStatus('error');
-        setError(err);
+        if (err.name !== 'AbortError') {
```
Comment (segdeha) [inline src/hooks/useLocations.js:29]: Is there a reliable way to generate this error so I can see what it looks like?

File: src/pages/MapPage.js [hunk 2]
```diff
@@ -13,18 +13,18 @@ const MapPage = () => {
 
   return (
     <>
-      <h1>What's Near Me?</h1>
-      <div className="view__content">
+      <div className="view__header">
+        <h1>What's Near Me?</h1>
         {status === 'loading' ? <Spinner /> : null}
```
Comment (segdeha) [inline src/pages/MapPage.js:18]: Does including the spinner inside the header cause a shift of content? Would it be possible instead to show it over the top of the map maybe?

File: src/pages/MapPage.js [hunk 3]
```diff
@@ -13,18 +13,18 @@ const MapPage = () => {
...
+      </div>
+      <div className="view__content">
         {status === 'error' ? <ErrorMessage error={error} /> : null}
-        {status === 'success' || status === 'cancelled' ? (
```
Comment (segdeha) [inline src/pages/MapPage.js:20]: Getting rid of this check makes the experience so much better!!! 👏👏👏

File: src/components/Map.js [hunk 4]
```diff
@@ -33,10 +33,7 @@ export const Map = ({
     const lat = event?.center?.lat();
     const lng = event?.center?.lng();
     const newMapCenter = { lat, lng };
-
-    window.setTimeout(() => {
-      setNewCenterMap(newMapCenter);
-    }, 1000);
+    setNewCenterMap(newMapCenter);
```
Comment (segdeha) [inline src/components/Map.js:36]: This feels better as an end-user, for sure! Do we have another mechanism for making sure we don't make multiple API calls at one time?

**Review by segdeha — APPROVED** [review]
I've left mostly questions (and kudos!), so I'm defaulting to approve on this. Would love for you to get @laurenmbeatty's input as well before merging (as well as making sure your teammates have seen and understand the change!). Thanks!

File: src/components/Map.js (hunk 4 above)
Comment (alxmcr) [reply src/components/Map.js:36]: Yes, there are two mechanisms (I think):

1. Our LocationsContext, we make sure that one call is executed in all our app, and share the results.
2. Using 'AbortController' in our custom hook `useLocations`, when we 'clean-up' this hook. if we're calling multiple times, they're canceled 'automatically'.

**Canceled requests**

image

File: src/hooks/useLocations.js (hunk 1 above)
Comment (alxmcr) [reply src/hooks/useLocations.js:29]: @segdeha: Yes! 

1. Open your Browser Dev Tools > Network.
2. You drag the map constantly then you can see requests canceled.

image

File: src/pages/MapPage.js (hunk 2 above)
Comment (alxmcr) [reply src/pages/MapPage.js:18]: @segdeha : I'm so sorry. I identified some visual issues, but I reduced the font-size to `h1`. I tested on some devices. I hope it's much better.

**Iphone 5/SE**

image

**Iphone 6/7/8 plus**

image


**Iphone 6/7/8**

image

File: src/hooks/useLocations.js (hunk 1 above)
Comment (laurenmbeatty) [reply src/hooks/useLocations.js:29]: Perfect...good call here, @alxmcr

File: src/pages/MapPage.js (hunk 2 above)
Comment (laurenmbeatty) [reply src/pages/MapPage.js:18]: I think eventually we may want to have the spinner just the very first time the map loads, and then do away with it, but I think this is a good compromise for now!

**Review by laurenmbeatty — APPROVED** [review]
Big improvement!

File: src/components/Map.js (hunk 4 above)
Comment (jssckbl) [reply src/components/Map.js:36]: This is nice and clean. I like it. And I was able to make it work to cancel an API call twice and watch based on this method. Thank you for asking, @segdeha and thank you for the screen shot, @alxmcr.

File: src/hooks/useLocations.js (hunk 1 above)
Comment (jssckbl) [reply src/hooks/useLocations.js:29]: This is nice and clean. I like it. And I was able to make it work to cancel an API call twice and watch based on this method. Thank you for asking, @segdeha and thank you for the screen shot, @alxmcr.

File: src/pages/MapPage.js [hunk 5]
```diff
@@ -13,18 +13,18 @@ const MapPage = () => {
 
   return (
     <>
-      <h1>What's Near Me?</h1>
-      <div className="view__content">
+      <div className="view__header">
+        <h1>What's Near Me?</h1>
```
Comment (jssckbl) [inline src/pages/MapPage.js:17]: @segdeha's comment made me wonder, should this `<h1>` be wrapped in a `<header>` for more semantic HTML?  I am not clear yet based on the semantic HTML and React docs that I've looked at so far. 

Definitely just a question, and not a block to merging!

**Review by jssckbl — APPROVED** [review]
This is helpful and makes sense to me.

Approved!

File: src/pages/MapPage.js (hunk 5 above)
Comment (alxmcr) [reply src/pages/MapPage.js:17]: Yes, Jessica! Thank you! I think you are right. I'm going to change to a `<header>` tag.

**Review by meganmckissack — APPROVED** [review]
This is so great @alxmcr!  The clean movement of the map without re-rendering is give my head and eyes a rest, thank you.

**Pull request merged**